backend/src/main/java/com/soen341/instagram/service/impl/RegistrationService.java

```diff
@@ -0,0 +1,62 @@
+package com.soen341.instagram.service.impl;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.exception.account.AccountExistException;
+import com.soen341.instagram.exception.account.InvalidEmailException;
+import com.soen341.instagram.model.Account;
+
+@Service("registrationService")
+public class RegistrationService
+{
+	@Autowired
+	private AccountRepository accountRepository;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	public void createNewAccount(final Account account) throws ParseException
+	{
+		if (accountRepository.findByUsername(account.getUsername()) == null
+				&& accountRepository.findByEmail(account.getEmail()) == null)
+		{
+			if (!isEmailFormatValid(account.getEmail()))
```

Comment:
We'll probably need to validate the username format as well

Reply:
Maybe this can be added in a later pass?

Reply:
I've added username validation
Character allowed are word, periods, numbers